go-controller/pkg/ovn/fake_address_set.go

```diff
@@ -0,0 +1,181 @@
+package ovn
+
+import (
+	"fmt"
+	"hash/fnv"
+	"net"
+	"strings"
+	"sync"
+
+	. "github.com/onsi/gomega"
+)
+
+func newFakeAddressSetFactory() *fakeAddressSetFactory {
+	return &fakeAddressSetFactory{
+		sets: make(map[string]*fakeAddressSet),
+	}
+}
+
+type fakeAddressSetFactory struct {
+	sync.RWMutex
+	// maps address set name to object
+	sets map[string]*fakeAddressSet
```

Comment:
dont need this or mutex anymore right?

Reply:
@trozet no, we need it for the fake one because there's no backing store (eg OVN) when doing ForEachAddressSet(). Though we could split that up and have a function for FakeAddressSetFactory to pre-seed the ForEachAddressSet() bits, but not the other ones.

But also, we need to run Eventually()/Expect() for the testcases, and that could be reading from the map at the same time that something is writing to it in the testcases. So I think we still need the mutex and the set map.